src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -82,6 +106,12 @@ public BindingMetadata BindingMetadata
                 if (_details.BindingMetadata == null)
                 {
                     var context = new BindingMetadataProviderContext(Identity, _details.ModelAttributes);
+
+                    // Provide a unique ModelBindingMessageProvider instance based on the one passed to the constructor
+                    // (from MvcOptions).
```

Comment:
You mean something like `Provide a unique ModelBindingMessageProvider instance so providers can customize it per-type.`?